pkg/operator/sync.go

```diff
@@ -195,6 +195,9 @@ func (optr *Operator) syncMutatingWebhook() error {
 func (optr *Operator) syncBaremetalControllers(config *OperatorConfig) error {
 	// Try to get baremetal provisioning config from a CR
 	baremetalProvisioningConfig, err := getBaremetalProvisioningConfig(optr.dynamicClient, baremetalProvisioningCR)
+	if err == nil && (BaremetalProvisioningConfig{}) == baremetalProvisioningConfig {
```

Comment:
Comparing the empty struct feels a bit odd. How big of a change would it be to make `getBaremetalProvisioningConfig` return a pointer?

Reply:
Done.